src/main/java/de/zalando/aruha/nakadi/FlowIdRequestFilter.java

```diff
@@ -0,0 +1,47 @@
+package de.zalando.aruha.nakadi;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class FlowIdRequestFilter implements Filter {
+    public static final String X_FLOW_ID_HEADER = "x-flow-id";
+
+    @Override
+    public void init(final FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+        FlowIdUtils.clear();
+        String flowId = null;
+
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+            flowId = httpServletRequest.getHeader(X_FLOW_ID_HEADER);
+        }
+
+        if (flowId == null) {
+            flowId = FlowIdUtils.generateFlowId();
```

Comment:
Let me guess, I think that it makes it easier to group interleaved logs as belonging to a single request?